frontend/src/app/sign-up/page.tsx

```diff
@@ -50,32 +50,31 @@ export default function Home() {
       return;
     }
 
-    // mock for backend
-    const res = await fetch("https://jsonplaceholder.typicode.com/user", {
-      method: "Post",
-      headers: {
-        Accept: "application/json, text/plain, */*",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        email: email,
-        password: password,
-      }),
+    await fetch(USER_API_ENDPOINT + "/register", {
+        method: "Post",
+        headers: {
+            Accept: "application/json, text/plain, */*",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            email: email,
+            password: password,
+            name: "Username placeholder",           // To be changed in the future
+            major: "User's major placeholder",      // To be changed in the future
+            course : "",                            // To be changed in the future
```

Comment:
It is good that you have course in the database schema, but i think we should allow them to have multiple courses, so u might need to convert this to array

Reply:
yeah course need to be changed to string[]